torch_ttnn/passes/lowering/to_tt_pass.py

```diff
@@ -768,12 +769,77 @@ def rewrite_node(node):
     return gm
 
 
+class ConstantFolder(torch.nn.Module):
```

Comment:
Can we move to a separate file? This one is growing way out of comfort zone

Reply:
Yeah, I think constant folding should be a separate compiler pass and in `backend.py` called before `ToTtPass`.

Reply:
I have moved to separate `ConstantFoldingPass` and apply before `ToTtPass`. refer to 4d3af2bead5d1da95e0ed3d4f601f00c62912024